packages/extension-driver-pg/src/lib/pgDataSource.ts

```diff
@@ -54,6 +54,13 @@ export class PGDataSource extends DataSource<any, PGOptions> {
     const { pool, options } = this.poolMapping.get(profileName)!;
     this.logger.debug(`Acquiring connection from ${profileName}`);
     const client = await pool.connect();
+    // listen the pg connection event errors when executing query
+    pool.on('error', (err) => {
+      //  Ignore the error and display warn message
+      this.logger.warn(
+        `Pool client of profile instance ${profileName} connecting failed, detail error, ${err}`
+      );
+    });
```

Comment:
NIT: Can we add some test cases for pool issues? e.g. Delete the docker container before connections closed.

Reply:
Hi @oscar60310, thanks for the suggestion, I have tried to add the test case by deleting the docker container before connections closed, but seems the test case could not assert the listener work of `pool.on('error', ...)`, or even I try to assert the `pool.emit('error', ...)`, it still not check.

Reply:
Here is my test code:
```ts

it('Data source should throw error when PG Server disconnected', async () => {
  // Arrange
  const pg = new PGServer();
  await pg.prepare();

  const dataSource = new PGDataSource({}, '', [
    {
      name: 'profile1',
      type: 'pg',
      connection: {
        host: pg.host,
        user: pg.user,
        password: pg.password,
        database: pg.database,
        port: pg.port,
      } as PGOptions,
      allow: '*',
    },
  ]);
  await dataSource.activate();

  const pool = dataSource.getPool('profile1');
  const spyEmit = sinon.default.spy(pool.emit);
  const spyOn = sinon.default.spy(pool.on);

  await pg.destroy();
  // Act
  const executeAction = dataSource.execute({
    statement: 'select * from users limit 1',
    bindParams: new Map(),
    profileName: 'profile1',
    operations: {} as any,
  });

  // Assert
  await expect(executeAction).rejects.toThrow();
  // Assert Pool Emit and On
  expect(spyEmit.calledWith('error', expect.any(Function))).toBeTruthy();
  expect(spyOn.calledWith('error' as any, expect.any(Function))).toBeTruthy();

  await dataSource.destroy();
  // afterEach hook will timeout if any leak occurred.
}, 150000);

```